### ISSUE TITLE
Firehose extended_s3 kms error

### ISSUE THREAD

**Issue by wsmithCTP** [terraform-providers/terraform-provider-aws#3133]
Hi there,

### Terraform Version
```
Terraform v0.11.2
+ provider.aws v1.7.1
+ provider.template v1.0.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_kinesis_firehose_delivery_stream

### Terraform Configuration Files
I cannot display them here.

### Expected Behavior
Resources should have been created normally with no console errors or state file problems.

### Actual Behavior
```* aws_kinesis_firehose_delivery_stream.firehose: extended_s3_configuration.0.kms_key_arn: '' expected type 'string', got unconvertible type 'firehose.KMSEncryptionConfig'```

When creating a kinesis firehose with the `extended_s3` destination/configuration, I am presented the above error. However, the resource is created and appears to be functioning, with the KMS key as an attribute in the console. The issue is that any subsequent `refresh`, `plan`, `apply`, and `destroy` are interrupted by this console error. I have tested it with the deprecated `s3` destination and I do not receive this error. 

### Steps to Reproduce
1. `Create a kinesis firehose with the extended_s3 destination with a kms key arn`
2. `terraform plan/apply/destroy`

**Comment by cs-a44319** [comment]
Getting same issue

**Comment by ericpsantos** [comment]
Same here, in version 0.11.3.

**Comment by gbergere** [comment]
Still here with :
- `terraform` -> `0.11.3` 
- `terraform-aws-provider` -> `1.8`

**Comment by ericpsantos** [comment]
@jbardin any ideas on this? Is there a workaround for now?

**Comment by dwoodlins** [comment]
Same issue. For us, the only "workaround" has been to revert to using version 1.6.x of terraform-aws-provider (1.7.0 had an unrelated blocking issue).

**Comment by pagemedias** [comment]
Can confirm get issue use terraform 0.11.2 with aws-provider 1.8

Using aws-provider 1.6 don’t get issue

**Comment by bflad** [comment]
Hi folks, sorry you were having trouble with this. There was not an acceptance test covering this functionality previously, which would have caught any regression. 😢  I have submitted a bug fix PR (#3301) that I'll try to get into the v1.9.0 release tomorrow.

**Comment by dwoodlins** [comment]
Great news, thanks!

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with v1.9.0 of the AWS provider, hopefully later today.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.9.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#190-february-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
